src/lib/viewers/media/DashViewer.js

```diff
@@ -279,12 +283,26 @@ class DashViewer extends VideoBaseViewer {
      */
     handleSubtitle() {
         const subtitleIdx = parseInt(this.cache.get('media-subtitles'), 10);
-        if (this.textTracks[subtitleIdx] !== undefined) {
+
+        // Auto-generated index 0 ==> turn auto-generated text track on
+        if (this.autoCaptionDisplayer && subtitleIdx === 0) {
+            // Manually set text visibility with the custom Shaka Text Displayer
+            this.autoCaptionDisplayer.setTextVisibility(true);
+            this.emit('subtitlechange', __('auto_generated'));
+
+            // Valid non-auto-generated index ==> turn specified text track on
```

Comment:
Move these comments into their blocks?

Reply:
I like having the comment above the if statement to describe the condition going into the block as well